backend/program/content/plex_watchlist.py

```diff
@@ -39,25 +40,28 @@ def run(self):
         items = self._get_items_from_plex_watchlist()
         new_items = [item for item in items if item not in self.media_items]
         container = self.updater.create_items(new_items)
+        for item in container:
+            item.set_requested_by("Plex Watchlist")
+        previous_count = len(self.media_items)
         added_items = self.media_items.extend(container)
-        if len(added_items) > 0:
-            logger.info("Added %s items", len(added_items))
+        added_items_count = len(self.media_items) - previous_count
+        if added_items_count != self.previous_added_items_count and added_items_count > 0:
+            logger.info("Added %s items", added_items_count)
+            self.previous_added_items_count = added_items_count
+        if added_items_count > 0:
+            for added_item in added_items:
+                logger.debug("Added %s", added_item.title)
 
     def _get_items_from_plex_watchlist(self) -> list:
         """Fetch media from Plex watchlist"""
-        response_obj = get(self.watchlist_url, timeout=5)
+        response_obj = get(self.watchlist_url, timeout=30)
         watchlist_data = json.loads(response_obj.response.content)
         items = watchlist_data.get("items", [])
         ids = []
         for item in items:
-            imdb_id = next(
-                (
-                    guid.split("//")[-1]
-                    for guid in item.get("guids")
-                    if "imdb://" in guid
-                ),
-                None,
-            )
-            ids.append(imdb_id)
-        logger.debug("Found %s items", len(ids))
+            imdb_id = next((guid.split("//")[-1] for guid in item.get("guids") if "imdb://" in guid), None)
+            if imdb_id:
+                ids.append(imdb_id)
+            else:
+                logger.debug("No imdb id found for %s", item.get("title"))

```

Comment:
Eventually will need to add tvdb conversion func to accommodate for this. In my testing I found that out of the 50 watchlist items I had, 3 of them were tvdb id's instead of imdb. None were tmdb.

Reply:
the guid attribute already includes tmdb and tvdb ids but this will do for now.

Reply:
May need your help with this from which guid you're talking about.. the item guid is always empty or filled with Plex's guid, is that the one your talking about?